src/compiler/program.ts

```diff
@@ -1862,6 +1866,9 @@ export function createProgram(rootNamesOrOptions: readonly string[] | CreateProg
         redirectTargetsMap,
         usesUriStyleNodeCoreModules,
         resolvedLibReferences,
+        getCurrentPackagesMap: () => packageMap,

```

Comment:
Is it possible for this call to occur before the package map has been initialized? Would we benefit from keeping the code from `checker.ts` which just memoizes it? Or, I guess just call` getPackagesMap()`?

Reply:
Hm, maybe it's intentionally not getting it; I see the `oldProgram` use below.

Reply:
this is to get it if present so that we can reuse the packageMap if program structure is same as old program., thats why not memoize